apps/ca.c

```diff
@@ -1708,8 +1709,17 @@ static int do_body(X509 **xret, EVP_PKEY *pkey, X509 *x509,
         goto end;
 
     /* Initialize the context structure */
-    X509V3_set_ctx(&ext_ctx, selfsign ? ret : x509,
-                   ret, req, NULL, X509V3_CTX_REPLACE);
+    if (!X509V3_set_ctx(&ext_ctx, selfsign ? ret : x509,
+                        ret, NULL, NULL, X509V3_CTX_REPLACE))
```

Comment:
It is not entirely clear to me why the `req` argument was replaced with NULL.  Is it redundant here?  Why?

Reply:
I removed it because otherwise `s2i_skey_id()` in `crypto/x509/v3_skid.c` may pick up the wrong value - 
here the one in `x509` is relevant.
Note that this PR now enforces and documents that at most one of the certificate (`subject`), the certification request (`req`), or the certificate revocation list (`crl`) parameters can be non-NULL.

Reply:
It'd be good to add a  brief comment that noted why the `req` argument is NULL, to avoid future regressions.

Reply:
One could do this, but meanwhile `X509V3_set_ctx()` would throw an error if `req` was given in addition to a reference cert or CRL.